config/deploy.rb

```diff
@@ -0,0 +1,111 @@
+lock '~> 3.17.1'
+
+set :use_sudo, false
+## STAGE_NAME is a paramter from Jenkins job: "staging", "qc", and "testing"
+set :stage_case, ENV['STAGE_NAME']
+
+set :application, 'curator_app'
+set :repo_url, "https://github.com/boston-library/#{fetch(:application)}.git"
+
+###### Make user home path dynamic.
+set :user, ENV['DEPLOY_USER']
+set :deploy_to, "/home/#{fetch(:user)}/railsApps/#{fetch(:application)}"
+
+set :rvm_installed, "/home/#{fetch(:user)}/.rvm/bin/rvm"
+set :rvm_ruby_version, File.read(File.expand_path('./../.ruby-version', __dir__)).strip
+set :rvm_bundle_version, File.read(File.expand_path('./Gemfile.lock'))[-10..-1].strip
+
+# Default value for :pty is false
+set :pty, true
+
+## As bin/puma, bin/pumactl are sensitive to current project directory, it is better not to use a symlink
+append :linked_files, 'config/database.yml', 'config/credentials/staging.key', 'config/credentials/production.key', 'config/environments/staging.rb'
+append :linked_dirs, 'log', 'tmp/cache', 'tmp/pids', 'tmp/sockets', 'bundle'
+
+# Default value for keep_releases is 5
+set :keep_releases, 5
+
+# Costomized tasks that restart our services
+namespace :boston_library do
+  desc 'Gem update'
+  task :gem_update do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} #{fetch(:rvm_ruby_version)} do gem update --system --no-document")
+    end
+  end
+
+  desc 'Install new ruby if ruby-version is required'
+  task :rvm_install_ruby do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} install #{fetch(:rvm_ruby_version)} -C --with-jemalloc")
+      execute("#{fetch(:rvm_installed)} use #{fetch(:rvm_ruby_version)}")
+    end
+  end
+
+  # desc 'Install bundler 2.3.26'
+  desc "Install bundler #{fetch(:rvm_bundle_version)}"
+  task :install_bundler do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} #{fetch(:rvm_ruby_version)} do gem install bundler:#{fetch(:rvm_bundle_version)}")
+    end
+  end
+
+  ## Update ruby version for systemd service
+  desc 'Update ruby version for systemd service'
+  task :update_service_ruby do
+    on roles(:app) do
+      execute("sudo rm /etc/systemd/system/\"#{fetch(:application)}\"_puma.service.d/override.conf | true
+              SERVICE_RUBY_VERSION=`cat /home/\"#{fetch(:user)}\"/railsApps/\"#{fetch(:application)}\"/current/.ruby-version`
+              echo \"SERVICE_RUBY_VERSION IS: \" ${SERVICE_RUBY_VERSION}
+              echo '[Service]' > override.conf
+              echo \"Environment=SERVICE_RUBY_VERSION=${SERVICE_RUBY_VERSION}\" >> override.conf
+              sudo mv override.conf /etc/systemd/system/\"#{fetch(:application)}\"_puma.service.d/override.conf
+              sudo /bin/systemctl daemon-reload")
+    end
+  end
+
+  desc 'Copy Gemfile and Gemfile.lock to shared directory'
+  task :upload_gemfile do
+    on roles(:app) do
+      %w[Gemfile Gemfile.lock].each do |f|
+        upload! ENV['PWD'] + '/' + f, "#{shared_path}/" + f
+      end
+    end
+  end
```

Comment:
This is a new task not present in the other apps we've worked on. Why is it needed?

Reply:
Capistrano deployment can be successful without this task.  So temporarily comment it out but keep in the script in case.